support convtranspose and activation checkpointing

guopengf opened the pull request:
https://github.com/Project-MONAI/GenerativeModels/issues/410 support ConvTranspose to avoid F.interpolate limitation and activation checkpointing to save memeory.

marksgraham reviewed (changes requested):
Hi @guopengf 

This looks good to me - thanks for adding. Do you think it would be possible to also implement the checkpointing in the VQ-VAE so we can maintain a consistent interface between the two?

guopengf commented:
Hi @marksgraham 

I think enabling checkpointing in the VQ-VAE is a good point. I implemented it and added corresponding tests. 

However, it is worth noting that currently, TorchScript does not support activation checkpointing. It will trigger the following `NotSupportedError` error:
`torch.jit.frontend.NotSupportedError: Compiled functions can't take variable number of arguments or use keyword-only arguments with defaults:
def checkpoint(function, *args, use_reentrant: bool = True, **kwargs):
                                                             ~~~~~~~ <--- HERE
    r"""Checkpoint a model or part of the model
'VQVAE.encode' is being compiled since it was called from 'VQVAE.forward'
  File "/home/pengfeig/code/GenerativeModels/generative/networks/nets/vqvae.py", line 440
    def forward(self, images: torch.Tensor) -> tuple[torch.Tensor, torch.Tensor]:
        quantizations, quantization_losses = self.quantize(self.encode(images))
                                                           ~~~~~~~~~~~~~~~~~~ <--- HERE
        reconstruction = self.decode(quantizations)
`
So I removed TorchScript test case [here](https://github.com/Project-MONAI/GenerativeModels/blob/d89e70306151511aa882825bb15f3e18a7f17756/tests/test_vqvae.py#L128).

I think for a similar reason, Walter removed TorchScript test case to support `xformers` package in AutoencoderKL. https://github.com/Project-MONAI/GenerativeModels/blob/20d517a7279774b90232de8bca107b65efe1f7b6/tests/test_autoencoderkl.py#L146
